iron-input.html

```diff
@@ -118,15 +118,25 @@
     },
 
     ready: function() {
-      this.bindValue = this.value;
+      if (this.type === 'number') {
+        this.bindValue = parseFloat(this.value);
+      } else {
+        this.bindValue = this.value;
+      }
     },
 
     /**
      * @suppress {checkTypes}
      */
     _bindValueChanged: function() {
-      if (this.value !== this.bindValue) {
-        this.value = !(this.bindValue || this.bindValue === 0) ? '' : this.bindValue;
+      if (this.type === 'number'){
+        if (parseFloat(this.value) !== this.bindValue) {
+          this.value = !(this.bindValue || this.bindValue === 0) ? 0 : this.bindValue;
```

Comment:
Can't we still assign `''` instead of `0`, and simplify the code a bit to
`if (this.value !== this.bindValue || parseFloat(this.value) !== this.bindValue) { ... }` 

(and maybe adding a comment why the latter bit is necessary for type="number" )